docs/contracts/cargo-contract.md

```diff
@@ -108,14 +108,17 @@ The final, optimized `flipper-pruned.wasm` file is what we will actually deploy
 ### Contract ABI
 
 By running the next command we'll generate the Application Binary Interface (ABI):
-``` bash
-cargo +nightly build --features ink-generate-abi
+
+```bash
+cargo +nightly run --package abi-gen
 ```
+The corresponding `abi.json` can be found in /target. Please note that for backward compatibility, a `old_abi.json` file is also generated. If the UI complains about an invalid ABI file, try the `old_abi.json` file.
```

Comment:
```suggestion
The corresponding `abi.json` can be found in `/target`. Please note that for backward compatibility, an `old_abi.json` file is also generated. If the UI complains about an invalid ABI file, try the `old_abi.json` file.
```

It is a little strange that the UI "might" complain. What is the situation where it will and wont?

Reply:
For now it will be complaining every time for `abi.json`. The new ABI is not yet supported on the UI. I wanted to keep this to keep it working at the time it becomes compatible with the UI.
This is why I'm just keeping it more open here and the contracts tutorial will have the exact instructions.